doc/sphinx/proof-engine/ltac.rst

```diff
@@ -1083,45 +1567,49 @@ Testing boolean expressions
    .. exn:: Condition not satisfied.
       :undocumented:
 
-Proving a subgoal as a separate lemma
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Proving a subgoal as a separate lemma: abstract
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-.. tacn:: abstract @ltac_expr
-   :name: abstract
+.. tacn:: abstract @ltac_expr2 {? using @ident }
 
-   From the outside, :n:`abstract @ltac_expr` is the same as :n:`solve @ltac_expr`.
-   Internally it saves an auxiliary lemma called ``ident_subproofn`` where
-   ``ident`` is the name of the current goal and ``n`` is chosen so that this is
-   a fresh name. Such an auxiliary lemma is inlined in the final proof term.
+   Does a :tacn:`solve` :n:`[ @ltac_expr2 ]` and saves the subproof as an auxiliary lemma.
+   if :token:`ident` is specified, the lemma is saved with that name; otherwise
+   the lemma is saved with the name `NAME_subproofN` where
+   `NAME` is the name of the current goal and `N` is
+   a fresh name. The auxiliary lemma is inlined in the final proof term.
 
-   This tactical is useful with tactics such as :tacn:`omega` or
-   :tacn:`discriminate` that generate huge proof terms. With that tool the user
-   can avoid the explosion at time of the Save command without having to cut
-   manually the proof in smaller lemmas.
+   This is useful with tactics such as :tacn:`omega` or
+   :tacn:`discriminate` that generate huge proof terms with many intermediate
+   goals.  It can significantly reduce peak memory use.  In most cases it doesn't
```

Comment:
```suggestion
   goals.  It can significantly reduce peak memory use at :cmd:`Qed`\-time.  In most cases it doesn't
```

I'm not sure of this. @JasonGross please confirm

Reply:
I think this is wrong.  I believe it significantly reduces peak memory usage across the entire proof script.